engine/api/application_variable.go

```diff
@@ -92,38 +108,43 @@ func (api *API) getVariablesInApplicationHandler() service.Handler {
 func (api *API) deleteVariableFromApplicationHandler() service.Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
 		vars := mux.Vars(r)
-		key := vars[permProjectKey]
+		projectKey := vars[permProjectKey]
 		appName := vars["applicationName"]
 		varName := vars["name"]
 
-		app, err := application.LoadByName(api.mustDB(), key, appName)
+		proj, err := project.Load(api.mustDB(), projectKey)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load project %s", projectKey)
+		}
+
+		app, err := application.LoadByProjectIDAndName(ctx, api.mustDB(), proj.ID, appName)
 		if err != nil {
-			return sdk.WrapError(err, "Cannot load application: %s", appName)
+			return sdk.WrapError(err, "cannot load application: %s", appName)
 		}
 		if app.FromRepository != "" {
 			return sdk.WithStack(sdk.ErrForbidden)
 		}
 
 		tx, err := api.mustDB().Begin()
 		if err != nil {
-			return sdk.WrapError(err, "Cannot start transaction")
+			return sdk.WrapError(err, "cannot start transaction")
 		}
 		defer tx.Rollback() // nolint
 
-		varToDelete, errV := application.LoadVariable(api.mustDB(), app.ID, varName)
-		if errV != nil {
-			return sdk.WrapError(errV, "deleteVariableFromApplicationHandler> Cannot load variable %s", varName)
+		varToDelete, err := application.LoadVariable(ctx, api.mustDB(), app.ID, varName)
```

Comment:
to load and delete a variable, we don't need to load the project, then application and finally the variable.
`projectKey`, `appName` and `varName` are identifiers

Reply:
Project is no more loaded